### PR TITLE
Keep incremental compiler state through full restart.

### PR DESCRIPTION
**Pull request by aam**
Resetting incremental compiler state is needed when it is used for restarts as well as reloads. But since we are currently use batch compiler for restarts, no need to reset incremental compiler at this point.

### CODE REVIEW

**Review by cbracken — APPROVED** [review]

File: packages/flutter_tools/lib/src/run_hot.dart [hunk 1]
```diff
@@ -335,14 +335,6 @@ class HotRunner extends ResidentRunner {
     }
 
     final Stopwatch restartTimer = new Stopwatch()..start();
-    if (previewDart2) {
```
Comment (a-siva) [inline packages/flutter_tools/lib/src/run_hot.dart]: Can you add a TODO here that you would have to add the reset back once the restart logic is switched back to using the incremental kernel generator.

File: packages/flutter_tools/lib/src/devfs.dart [hunk 2]
```diff
@@ -435,6 +442,13 @@ class DevFS {
       String archivePath;
       if (deviceUri.path.startsWith(assetBuildDirPrefix))
         archivePath = deviceUri.path.substring(assetBuildDirPrefix.length);
+      // When doing full restart in preview-dart-2 mode, copy content so
```
Comment (a-siva) [inline packages/flutter_tools/lib/src/devfs.dart]: Similarly a TODO here that this clone will not be needed once we switch hot restart to use the IKG

**Review by a-siva — APPROVED** [review]

File: packages/flutter_tools/lib/src/run_hot.dart (hunk 1 above)
Comment (aam) [reply packages/flutter_tools/lib/src/run_hot.dart]: Done

File: packages/flutter_tools/lib/src/devfs.dart (hunk 2 above)
Comment (aam) [reply packages/flutter_tools/lib/src/devfs.dart]: Done

**Comment by aam** [comment]
@cbracken with this landed all benchmarks except `hotRestartMillisecondsToFrame` went back to previous(normal) values, so I think we can reset the baseline for restart one for now.

**Comment by cbracken** [comment]
Done. Thanks!

**Pull request merged**